llvm/lib/Target/SPIRV/SPIRVModuleAnalysis.cpp

```diff
@@ -633,7 +633,8 @@ void RequirementHandler::initAvailableCapabilities(const SPIRVSubtarget &ST) {
   if (ST.isAtLeastSPIRVVer(VersionTuple(1, 6)))
     addAvailableCaps({Capability::DotProduct, Capability::DotProductInputAll,
                       Capability::DotProductInput4x8Bit,
-                      Capability::DotProductInput4x8BitPacked});
+                      Capability::DotProductInput4x8BitPacked,
+                      Capability::DemoteToHelperInvocation});
```

Comment:
this doesn't look right. Docs say this should be available in SPIRV 1.0 https://github.com/KhronosGroup/SPIRV-Registry/blob/main/extensions/EXT/SPV_EXT_demote_to_helper_invocation.asciidoc

Reply:
This looks like a spec bug [OpDemoteToHelperInvocationEXT was updated to 1.6](https://github.com/KhronosGroup/SPIRV-Headers/blob/996c728cf7dcfb29845cfa15222822318f047810/include/spirv/unified1/spirv.core.grammar.json#L6475 ):
with this pr: https://github.com/KhronosGroup/SPIRV-Headers/pull/259